python/python/lance/dataset.py

```diff
@@ -38,6 +38,16 @@ def __init__(self, uri: Union[str, Path], version: Optional[int] = None):
         self._uri = uri
         self._ds = _Dataset(uri, version)
 
+    def __reduce__(self):
+        return LanceDataset, (self.uri, self._ds.version())
+
+    def __getstate__(self):
+        return self.uri, self._ds.version()
+
+    def __setstate__(self, state):
+        self._uri, version = state
```

Comment:
This is the fix? does polas have issues with picking / distributed pickle to different machines? I am not certain now, whether `_Dataset` has any OS resource that can not be pickled / unpickled from another machine.

Not a blocker tho.